addons/sale_timesheet/controllers/portal.py

```diff
@@ -76,16 +107,28 @@ def _task_get_page_view_values(self, task, access_token, **kwargs):
         try:
             if task.sale_order_id and self._document_check_access('sale.order', task.sale_order_id.id):
                 values['so_accessible'] = True
+                title = _('Quotation') if task.sale_order_id.state in ['draft', 'sent'] else _('Sales order')
+                values['task_link_section'].append({
+                    'access_url': task.sale_order_id.get_portal_url(),
+                    'title': title,
+                })
         except (AccessError, MissingError):
             pass
 
-        values['invoices_accessible'] = []
-        for invoice in task.sale_order_id.invoice_ids:
-            try:
-                if self._document_check_access('account.move', invoice.id):
-                    values['invoices_accessible'].append(invoice.id)
-            except (AccessError, MissingError):
-                pass
+        if task.sale_order_id.invoice_ids:
+            moves = request.env['account.move'].search([('id', 'in', task.sale_order_id.invoice_ids.ids)])
```

Comment:
You already have the invoices with `task.sale_order_id.invoice_ids`, so why you did a search? 🤔

Reply:
There might be a case where the user does not have rights to access all the invoices, in order to get the records which user has rights to we did a search.